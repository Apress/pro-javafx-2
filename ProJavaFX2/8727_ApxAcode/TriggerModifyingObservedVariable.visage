/*
 * @test
 * @run
 */

// Using a trigger to keep the value of i to be between 0 and 9999.
var i = 1024 on replace {
  println("Variable i changed to {i}.");
  if (i < 0) {
    println("Since {i} < 0, setting i to 0.");
    i = 0;
  } else if (i > 9999) {
    println("Since {i} > 9999, setting i to 9999.");
    i = 9999;
  }
};

i = -100;
i = 20000;
