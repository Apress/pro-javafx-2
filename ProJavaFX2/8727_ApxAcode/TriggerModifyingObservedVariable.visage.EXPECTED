Variable i changed to 1024.
Variable i changed to -100.
Since -100 < 0, setting i to 0.
Variable i changed to 0.
Variable i changed to 20000.
Since 20000 > 9999, setting i to 9999.
Variable i changed to 9999.
